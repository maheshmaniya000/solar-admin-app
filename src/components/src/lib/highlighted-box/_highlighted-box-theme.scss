@use '~@angular/material' as mat;

@mixin theme($theme-or-color-config) {
  $color-config: mat.get-color-config($theme-or-color-config);
  $background: map-get($color-config, background);
  $primary: map-get($color-config, primary);

  sg-highlighted-box {
    background-color: mat.get-color-from-palette($background, app-bar);

    .title {
      color: mat.get-color-from-palette($primary);
    }
  }
}
