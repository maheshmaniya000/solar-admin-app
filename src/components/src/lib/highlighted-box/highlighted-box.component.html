<h4 class="title" *ngIf="title">{{ title }}</h4>
<ng-content></ng-content>
