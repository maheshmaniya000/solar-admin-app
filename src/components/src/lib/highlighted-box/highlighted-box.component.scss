:host {
  display: block;
  padding: 24px;
  border-radius: 4px;
}

.title {
  text-align: center;
}
