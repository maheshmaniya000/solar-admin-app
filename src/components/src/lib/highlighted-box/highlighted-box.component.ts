import { ChangeDetectionStrategy, Component, Input } from '@angular/core';

@Component({
  selector: 'sg-highlighted-box',
  templateUrl: './highlighted-box.component.html',
  styleUrls: ['./highlighted-box.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class HighlightedBoxComponent {
  @Input() title: string;
}
