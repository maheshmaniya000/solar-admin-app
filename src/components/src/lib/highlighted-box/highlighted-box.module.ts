import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';

import { HighlightedBoxComponent } from './highlighted-box.component';

@NgModule({
  declarations: [HighlightedBoxComponent],
  imports: [CommonModule],
  exports: [HighlightedBoxComponent]
})
export class HighlightedBoxModule {}
