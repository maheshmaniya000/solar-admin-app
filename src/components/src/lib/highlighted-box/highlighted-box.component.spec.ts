import { createHostFactory, SpectatorHost } from '@ngneat/spectator/jest';

import { HighlightedBoxComponent } from './highlighted-box.component';

describe('HighlightedBoxComponent', () => {
  describe('component', () => {
    const createHost = createHostFactory({
      component: HighlightedBoxComponent,
      shallow: true
    });
    let spectator: SpectatorHost<HighlightedBoxComponent>;

    beforeEach(() => {
      spectator = createHost(`
        <sg-highlighted-box>
          <button>Action</button>
        </sg-highlighted-box>
      `);
    });

    it('should match snapshot with transcluded content and without title ', () => {
      expect(spectator.fixture).toMatchSnapshot();
    });

    it('should match snapshot with title and transcluded content', () => {
      spectator.setInput('title', 'some title');
      expect(spectator.fixture).toMatchSnapshot();
    });
  });
});
