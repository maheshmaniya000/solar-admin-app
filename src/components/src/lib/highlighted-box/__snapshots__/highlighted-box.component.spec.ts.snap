// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HighlightedBoxComponent component should match snapshot with title and transcluded content 1`] = `
<ng-component
  __ngContext__={[Function LRootView]}
>
  <sg-highlighted-box>
    <h4
      class="title"
    >
      some title
    </h4>
    <button>
      Action
    </button>
  </sg-highlighted-box>
</ng-component>
`;

exports[`HighlightedBoxComponent component should match snapshot with transcluded content and without title  1`] = `
<ng-component
  __ngContext__={[Function LRootView]}
>
  <sg-highlighted-box>
    <button>
      Action
    </button>
  </sg-highlighted-box>
</ng-component>
`;
